spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/validation/ValidatorAdapter.java

```diff
@@ -153,4 +154,20 @@ private static Validator wrap(Validator validator, boolean existingBean) {
 		return validator;
 	}
 
+	@Override
+	@SuppressWarnings("unchecked")
+	public <T> T unwrap(Class<T> type) {
+		if (this.target == null) {
+			return null;
+		}
+		if (type.isAssignableFrom(this.target.getClass())) {
+			if (this.target instanceof SpringValidatorAdapter adapter) {
+				return adapter.unwrap(type);
+			}
+			return (T) this.target;
+		}
+
+		throw new ValidationException("Cannot unwrap to " + type);
```

Comment:
I think an `IllegalStateException` might be better here and perhaps the error message should include `target`. Something like `"Cannot unwrap " + target + " to " + type.getName()`

Reply:
Went with an IllegalArgumentException when given a `type` a `SmartValidator` cannot be unwrapped to as it seemed more sound to me?